open_rarity/scoring/token_feature_extractor.py

```diff
@@ -0,0 +1,42 @@
+from open_rarity.models import Token
+from open_rarity.models.collection import Collection
+from open_rarity.models.token_features import TokenFeatures
+
+
+class TokenFeatureExtractor:
+    """
+    Utility class that extract features from tokens
+    """
+
+    @staticmethod
+    def extract_unique_attribute_count(
+        token: Token, collection: Collection
+    ) -> TokenFeatures:
+        """This method extracts unique attributes count from the token
+
+        Parameters
+        ----------
+        token : Token
+            The token to extract features from
+        collection : Collection
+            The collection with the attributes frequency counts to base the
+            token trait probabilities on to calculate score.
+
+        Returns
+        -------
+        TokenFeatures
+            Token features wrapper class that contains extracted features
+
+        """
+
+        unique_attributes_count: int = 0
+
+        for _, string_attribute in token.metadata.string_attributes.items():
```

Comment:
Probably no need to do items again.
`for string_attributes in token.metadata.string_attributes.values():`